test/DiTest.php

```diff
@@ -1017,4 +1025,102 @@ public function testGetInstanceWithParamsHasSameNameAsDependencyParam()
         $di = new Di(null, null, $config);
         $this->assertCount(1, $di->get('ZendTest\Di\TestAsset\AggregatedParamClass')->aggregator->items);
     }
+
+    /**
+     * constructor injection consultation
+     *
+     * @group 4714
+     * @group 6388
+     */
+    public function testResolveMethodParametersDifferentParams()
```

Comment:
This test seems to be very complex. Can it be reduced into multiple smaller tests, or is it just like this?